adal/src/main/java/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -843,6 +843,17 @@ private String checkInputParameters(final String resource, final String clientId
         }
 
         final String redirectUri;
+
+        //check the permissions required for the broker usage
+        if (AuthenticationSettings.INSTANCE.getUseBroker()) {
```

Comment:
Move to method named "CheckForPermissionAPI22orLess". This should do the following:
1. if API < 23 && useBroker == true && permissionNotGranted, then return error. Here by return error, throw UsageAuthenticationException and pass it to callback
2. Wrap calls to checkInputParameters and checkForPermissionAPI22orLess into a wrapper function checkPreReqs and replace checkInputParameters